### ISSUE TITLE
An exception is raised when :max-queued-requests is lower than :min-thread when creating a jetty server

### ISSUE THREAD

**Issue by Akeboshiwind** [ring-clojure/ring#354]
I've just updated ring-jetty-adapter from 1.6.3 to 1.7.1 and previously I had :max-queued-requests set to 0 while :max-threads and :min-threads where both set to 64. This now raises an IllegalArgumentException for BlockingArrayQueue, changing :max-queued-requests to 64 stops the error from appearing.

Would it be possible to Raise an exception with a more useful error message, for example ":min-queued-requests must be greater than or equal to :min-threads"?
Or possibly the value of :max-queued-requests could be set to :min-threads if it is less than :min-threads.

**Comment by Akeboshiwind** [comment]
I've created a simple example of the issue here: https://github.com/Akeboshiwind/jetty-error-example/blob/master/src/jetty_error_example/core.clj

**Comment by weavejester** [comment]
Yes, I'd accept a PR for that.

**Comment by Akeboshiwind** [comment]
Which option would you prefer?
Raising a more useful exception or resetting the value of :max-queued-requests?

I think the second would be better seeing as it requires less knowledge for the user about the jetty-adapter's internals, although it does ignore the setting they've passed in.

**Comment by weavejester** [comment]
Ah, sorry, I meant for raising a more informative error message.

**Comment by Akeboshiwind** [comment]
What's happening with this issue then?
Will you be making an update to the library when you have time?

**Comment by weavejester** [comment]
Yes, I'll be making an update to Ring when I have time, but this is pretty far down on my list of priorities.

**Comment by Akeboshiwind** [comment]
Cool, thanks for the update :)
